### ISSUE TITLE
:case-symbol-test should warn on symbol in list

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2295]
```
~/dev/clj-kondo (master) $ clj -M:clj-kondo/dev --lint - --config '{:linters {:case-symbol-test {:level :warning}}}' <<< '(case 1 z 2)'
<stdin>:1:9: warning: Case test symbol is compile time constant and is never evaluated.
linting took 71ms, errors: 0, warnings: 1
~/dev/clj-kondo (master) $ clj -M:clj-kondo/dev --lint - --config '{:linters {:case-symbol-test {:level :warning}}}' <<< '(case 1 (z x) 2)'
linting took 37ms, errors: 0, warnings: 0
```

cc @IGJoshua
